### ISSUE TITLE
twisted.test.test_internet.ReactorCoreTestCase defines a timeout method which conflicts with trial's timeout attribute

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2056]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2056|
|Type|defect|
|Created|2006-09-02 20:23:52Z|

This should be fixed.

<details><summary>Searchable metadata</summary>

```
trac-id__2056 2056
type__defect defect
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1157228632000000 1157228632000000
changetime__1157565740000000 1157565740000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Ready for review in remove-timeout-method-2056

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)<a name="note_3"></a>|@jerub set owner to @exarkun|
|-|-|

There's a heap of changes here, but the tests pass, so I'm trusting that this does what is intended. Thanks for cleaning up the docstrings and the spacing. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [18063]) Merge remove-timeout-method-2056

Author: exarkun
Reviewer: jerub
Fixes [#2056](https://github.com/twisted/twisted/issues/2056)

This removes the timeout method from a twisted.internet TestCase which conflicted
with trial's timeout feature and caused a warning to be emitted.

Several nearby tests are trivially cleaned up and a docstring has also been added.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
